Lesson4/vknews/vknews/Controllers/LogInController/LogInViewController.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  LogInViewController.swift
+//  vknews
+//
+//  Created by itisioslab on 04.10.2021.
+//
+
+import UIKit
+
+class LoginViewController: UIViewController {
+
+    //    MARK: - IBOutlets
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    
+    //    MARK: - Properties
+    let validator: Validator = Validator()
+    let userService = UserService()
+    
+    // MARK: - Life Cycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        validator.delegate = self
+    }
+
+    //    MARK: - IBAction
+    @IBAction func logInAction(_ sender: Any) {
+        guard let tabBarViewController = storyboard?.instantiateViewController(withIdentifier: "TabBarViewController") as? TabBarViewController else { return }
+        
+        guard let email = emailTextField.text else {
+            return
+        }
+        
+        guard let password = passwordTextField.text else {
+            return
+        }
+
+        if !validator.validateEmail(email) { return }
+        
+        if !validator.validatePassword(password) { return }
+        
+        else if let currentUser = userService.getUser(email: email, password: password) {
```

Comment:
```suggestion
        } else if let currentUser = userService.getUser(email: email, password: password) {
```
обычно так делается чтобы понятно было что закрывает